src/components/SeeResults/SeeResults.tsx

```diff
@@ -384,32 +391,260 @@ function SeeResults(): JSX.Element {
 	}, [resultsView.round])
 
 	const changeFrame = runTransaction(changeFrameTX)
+	const Phase = Phases[resultsView.phase]
+	const Keyframe = Keyframes[resultsView.frame[0]]
+	const state = resultsView.frame[1]
 	return (
 		<div className={scss.class}>
-			{electionRef.current ? <ElectionRounds election={electionRef.current} /> : null}
-			<div>{resultsView.round}</div>
-			<div>{resultsView.phase}</div>
-			<div>{resultsView.frame}</div>
-			<button
-				type="button"
-				disabled={viewLocation === `beginning`}
-				onClick={() => {
-					changeFrame(`prev`)
-				}}
-			>
-				prev
-			</button>
-			<button
-				type="button"
-				disabled={viewLocation === `end`}
-				onClick={() => {
-					changeFrame(`next`)
-				}}
-			>
-				next
-			</button>
+			<main>
+				<header>{resultsView.round}</header>
+				<main>
+					<LayoutGroup>
+						{electionRef.current ? (
+							<Phase election={electionRef.current}>
+								<Keyframe state={state} election={electionRef.current} />
+							</Phase>
+						) : (
+							<div data-keyframe="loading">
+								<header>loading</header>
+							</div>
+						)}
+					</LayoutGroup>
+				</main>
+			</main>
+			<nav>
+				<button
+					type="button"
+					disabled={viewLocation === `beginning`}
+					onClick={() => {
+						changeFrame(`prev`)
+					}}
+				>
+					prev
+				</button>
+				<button
+					type="button"
+					disabled={viewLocation === `end`}
+					onClick={() => {
+						changeFrame(`next`)
+					}}
+				>
+					next
+				</button>
+			</nav>
+			<aside>
+				{electionRef.current ? <ElectionRounds election={electionRef.current} /> : null}
+			</aside>
 		</div>
 	)
 }
 
+export const Phases = {
+	surplus_allocation({ children }) {
```

Comment:
why `snake_case` here?

Reply:
good q. This is generally how I format magic strings. However, these became a lot more than that...